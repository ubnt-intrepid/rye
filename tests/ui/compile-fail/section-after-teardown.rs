fn main() {}

#[rye::test]
fn section_after_teardown(cx: &mut rye::Context<'_>) {
    println!("startup1");
    println!("startup2");

    section!(cx, "section1", {});
    section!(cx, "section2", {});

    println!("teardown1");
    println!("teardown2");

    section!(cx, "section3", {});
}

#[rye::test]
fn section_after_teardown_nested(cx: &mut rye::Context<'_>) {
    section!(cx, "root", {
        println!("startup1");
        println!("startup2");

        section!(cx, "section1", {});
        section!(cx, "section2", {});

        println!("teardown1");
        println!("teardown2");

        section!(cx, "section3", {});
    });
}
