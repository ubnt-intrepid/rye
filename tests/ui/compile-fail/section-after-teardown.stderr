error: section cannot be described after teardown
  --> $DIR/section-after-teardown.rs:14:5
   |
14 |     section!(cx, "section3", {});
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: section cannot be described after teardown
  --> $DIR/section-after-teardown.rs:29:9
   |
29 |         section!(cx, "section3", {});
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
